ksqldb-common/src/main/java/io/confluent/ksql/security/oauth/ClientSecretIdpConfig.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2025 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.security.oauth;
+
+import io.confluent.ksql.security.AuthType;
+import io.confluent.ksql.security.KsqlClientConfig;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public final class ClientSecretIdpConfig implements IdpConfig {
+  private String idpTokenEndpointUrl = "";
+  private String idpClientId = "";
+  private String idpClientSecret = "";
+  private String idpScope = "";
+  private String idpScopeClaimName = "scope";
+  private String idpSubClaimName = "sub";
+  private Short idpCacheExpiryBufferSeconds = 300;
+
+  @Override
+  public AuthType getAuthType() {
+    return AuthType.OAUTHBEARER;
+  }
+
+  @Override
+  public String getAuthenticationMethod() {
+    return "CLIENT_SECRET";
+  }
+
+  @Override
+  public Map<String, Object> getIdpConfigs() {
+    final Map<String, Object> mp = new HashMap<>();
```

Comment:
nit: name `mp` better.
```suggestion
    final Map<String, Object> mp = new HashMap<>();
```